core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -2441,6 +2637,42 @@ static bool
 shift_imm_imm_to_r_i32(x86::Assembler &a, SHIFT_OP op, int32 reg_no_dst,
                        int32 data1_src, int32 data2_src)
 {
+    int32 data;
+    Imm imm;
+
+    switch (op) {
+        case SHL:
+        {
+            data = data1_src << data2_src;
+            break;
+        }
+        case SHRS:
+        {
+            data = data1_src >> data2_src;
+            break;
+        }
+        case SHRU:
+        {
+            data = data1_src >> data2_src;
+            if (data & (1 << 32)) {
+                /* signed */
+                uint32 mask = (1 << (32 - data2_src)) - 1;
+                /* patch sign bits */
+                data = data & mask;
```

Comment:
How about: `data = (uint32)data1_src >> (uint32)data2_src;`

Reply:
Sure we can. BTW, shall we make sure `int32 data2_src` is always `>=0` ?

Reply:
The definition of wasm opcode SHR_U:
ishr_u𝑁(𝑖1, 𝑖2)
• Let 𝑗2 be 𝑖2 modulo 𝑁.
• Return the result of shifting 𝑖1 right by 𝑗2 bits, extended with 0 bits.

In fact here we should mod data_src2 with 32 firstly, but seems that in x86, shift with an operand larger than 32 equals to shift with (operand % 32).

In classic interpreter, we return `data = (uint32)data1_src >> ((uint32)data2_src % 32)`, see:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_interp_classic.c#L2274
and
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_interp_classic.c#L418-L423

Reply:
Sure. will `mod right` in IR first.